BaseTools/Plugin/OverrideValidation/OverrideValidation.py

```diff
@@ -628,20 +657,61 @@ def path_parse():
     dummy_list = []
     pathtool = Edk2Path(Paths.WorkSpace, dummy_list)
 
-    # Use absolute module path to find package path
-    pkg_path = pathtool.GetContainingPackage(Paths.TargetPath)
-    rel_path = Paths.TargetPath[Paths.TargetPath.find(pkg_path):]
+    # check if we are asked to update an .inf file "in-place"
+    if (Paths.RegenPath is not None):
+        if (Paths.RegenPrefix is not None):
+            path_prefixes = Paths.RegenPrefix
+        else:
+            path_prefixes = []
+        path_prefixes.append (Paths.WorkSpace)
+
+        v1_regex = re.compile(r"#(Override|Track) : (.*?) \| (.*?) \| (.*?) \| (.*?)")
+        v2_regex = re.compile(r"#(Override|Track) : (.*?) \| (.*?) \| (.*?) \| (.*?) \| (.*?)")
+        with open (Paths.RegenPath) as fd:
+            RegenInfData = fd.read()
+
+        RegenInfOutData = ""
+        for line in RegenInfData.splitlines (True):
+            match = v1_regex.match(line)
+            if match is None:
+                match = v2_regex.match(line)
+
+            if match is not None:
+                rel_path = match.group(3)
+                abs_path = resolvePath(path_prefixes, rel_path)
+
+                mod_hash = ModuleHashCal(abs_path)
+                # only update the line if the hash has changed - this ensures the timestamp tracks actual changes rather than last time it was run.
+                if (mod_hash != match.group(4)):
+                    VERSION_INDEX = Paths.Version - 1
+
+                    if VERSION_INDEX == 0:
+                        line = '#%s : %08d | %s | %s | %s\n' % (match.group(1), FORMAT_VERSION_1[0], rel_path, mod_hash, datetime.utcnow().strftime("%Y-%m-%dT%H-%M-%S"))
+                    elif VERSION_INDEX == 1:
+                        git_hash = ModuleGitHash(abs_path)
```

Comment:
Could this abs_path potentially come from a track tag, that does not exist in the current repo configuration? Maybe we should check it?

Reply:
well, it has to exist in order for the abs_path to resolve. So if you want to use the tool in this mode, all the paths referenced in the target .inf need to exist. I could modify the code to print a warning and skip if the path doesn't exist, is that more desirable?

Reply:
done.